arm-insights/2016-03-01/swagger/insightsManagementClient_LogProfiles.json

```diff
@@ -0,0 +1,268 @@
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to delete a log profile"
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "LogProfiles"
+        ],
+        "operationId": "LogProfiles_Get",
+        "description": "Gets the log profile.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/LogProfileNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get more information about a log profile.",
+            "schema": {
+              "$ref": "#/definitions/LogProfileResource"
+            }
+          }
+        }
+      },
+      "put": {
+        "tags": [
+          "LogProfiles"
+        ],
+        "operationId": "LogProfiles_CreateOrUpdate",
+        "description": "Create or update the log profile.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/LogProfileNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/LogProfileResource"
+            },
+            "description": "Parameters supplied to the operation."
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to create or update a log profile",
+            "schema": {
+              "$ref": "#/definitions/LogProfileResource"
+            }
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/microsoft.insights/logprofiles": {
+      "get": {
+        "tags": [
+          "LogProfiles"
+        ],
+        "operationId": "LogProfiles_List",
+        "description": "List the log profiles.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to list log profiles",
+            "schema": {
+              "$ref": "#/definitions/LogProfileCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": null
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Resource": {
+      "properties": {
+        "id": {
+          "type": "string",
+          "readOnly": true,
+          "description": "Azure resource Id"
+        },
+        "name": {
```

Comment:
With a simple fix on the service side, the name can be made a readOnly property. It wouldn't be a breaking change. I would encourage you to do that.